File: KnowMyBoard/app/src/main/java/com/huawei/hms/knowmyboard/java/dtse/activity/util/MySharedPreferences.java
/*
 * Copyright (c) Huawei Technologies Co., Ltd. 2020-2022. All rights reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.huawei.hms.knowmyboard.dtse.activity.util;

import android.content.Context;
import android.content.SharedPreferences;

public class MySharedPreferences {
    public static final String TAG = "MySharedPreferences";
    private SharedPreferences myPreferences;
    private SharedPreferences.Editor myEditor;
    private static MySharedPreferences mySharedPreferences;


    public MySharedPreferences(Context context) {
        myPreferences = context.getSharedPreferences(TAG, Context.MODE_PRIVATE);
        myEditor = myPreferences.edit();
    }

    public static MySharedPreferences getInstance(Context context) {
        if (mySharedPreferences == null) {
            synchronized (MySharedPreferences.class) {
                if(mySharedPreferences == null) {
                    mySharedPreferences = new MySharedPreferences(context);
                }
            }
        }
        return mySharedPreferences;
    }

    public void putStringValue(String key, String value) {
        myEditor.putString(key, value);
        myEditor.commit();
    }

    public String getStringValue(String key) {
        return myPreferences.getString(key, Constants.USER_NAME);
    }

}

File: KnowMyBoard/app/src/main/java/com/huawei/hms/knowmyboard/kotlin/dtse/activity/util/MySharedPreferences.kt
package com.huawei.hms.knowmyboard.dtse.activity.util

import android.content.Context
import android.content.SharedPreferences
import com.huawei.hms.knowmyboard.dtse.activity.util.MySharedPreferences

class MySharedPreferences(context: Context) {
    private val myPreferences: SharedPreferences
    private val myEditor: SharedPreferences.Editor
    fun putStringValue(key: String?, value: String?) {
        myEditor.putString(key, value)
        myEditor.commit()
    }

    fun getStringValue(key: String?): String? {
        return myPreferences.getString(key, Constants.USER_NAME)
    }

    companion object {
        const val TAG = "MySharedPreferences"
        private var mySharedPreferences: MySharedPreferences? = null
        @JvmStatic
        fun getInstance(context: Context): MySharedPreferences? {
            if (mySharedPreferences == null) {
                synchronized(MySharedPreferences::class.java) {
                    if (mySharedPreferences == null) {
                        mySharedPreferences = MySharedPreferences(context)
                    }
                }
            }
            return mySharedPreferences
        }
    }

    init {
        myPreferences = context.getSharedPreferences(TAG, Context.MODE_PRIVATE)
        myEditor = myPreferences.edit()
    }
}